frontend/src/sagas.js

```diff
@@ -171,6 +168,43 @@ import { getExchangeRates } from "./getExchangeRates";
 
 const api = new Api();
 
+// SELECTORS
+const getSelfId = state => {
+  return state.getIn(["login", "id"]);
+};
+const getJwt = state => state.toJS().login.jwt;
+const getEnvironment = state => {
+  const env = state.getIn(["login", "environment"]);
+  if (env) {
+    return env;
+  }
+  return "Test";
+};
+const getProjectHistoryState = state => {
+  return {
+    currentHistoryPage: state.getIn(["detailview", "currentHistoryPage"]),
+    lastHistoryPage: state.getIn(["detailview", "lastHistoryPage"]),
+    historyPageSize: state.getIn(["detailview", "historyPageSize"]),
+    totalHistoryItemCount: state.getIn(["detailview", "totalHistoryItemCount"])
+  };
+};
+const getSubprojectHistoryState = state => {
+  return {
+    currentHistoryPage: state.getIn(["workflow", "currentHistoryPage"]),
+    lastHistoryPage: state.getIn(["workflow", "lastHistoryPage"]),
+    historyPageSize: state.getIn(["workflow", "historyPageSize"]),
+    totalHistoryItemCount: state.getIn(["workflow", "totalHistoryItemCount"])
+  };
+};
+const getWorkflowitemHistoryState = state => {
+  return {
+    currentHistoryPage: state.getIn(["workflowitemDetails", "currentHistoryPage"]),
+    lastHistoryPage: state.getIn(["workflowitemDetails", "lastHistoryPage"]),
```

Comment:
Maybe we can remove this, since we are not using it in the sagas

Reply:
removed